### PR TITLE
AP723 Dependants Income

### PR DESCRIPTION
**Pull request by Obsiye**
Dependants income page, which records whether the dependant has an income or not. If it does then record this value

[Link to story](https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=233&projectKey=AP&modal=detail&selectedIssue=AP-723)

- Add attributes/columns to the Dependants table: has_income and monthly_income
- Create controller, form, view and route for Dependants income get and patch paths
- Update locales
- Update Citizen dependant flow path
- Create request and feature tests

## Checklist

Before you ask people to review this PR:

- [X] Tests and rubocop should be passing: `bundle exec rake`
- [X] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [X] There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- [X] The PR description should say what you changed and why, with a link to the JIRA story.
- [X] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [X] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/services/flow/flows/citizen_dependants.rb [hunk 1]
```diff
@@ -15,13 +15,17 @@ class CitizenDependants < FlowSteps
...
+          forward: ->(_, dependant) { dependant.adult_relative? || dependant.eighteen_or_less? ? :dependants_monthly_incomes : :dependants_education }
         },
         dependants_education: {
           path: ->(_, dependant) { "[PLACEHOLDER] #{dependant.name} dependants education" }
         },
-        dependants_income: {
-          path: ->(_, dependant) { "[PLACEHOLDER] #{dependant.name} dependants income" }
+        dependants_monthly_incomes: {
+          path: ->(_, dependant) { urls.citizens_dependant_monthly_income_path(dependant) },
+          forward: ->(_) { :dependants_assets }
```
Comment (Kalagan) [inline app/services/flow/flows/citizen_dependants.rb]: you can simply write `forward: :dependants_assets` here.

File: db/schema.rb [hunk 2]
```diff
@@ -308,25 +310,25 @@
...
-    t.decimal "trust_value", precision: 14, scale: 2
+    t.decimal "second_home_value"
+    t.decimal "second_home_mortgage"
+    t.decimal "second_home_percentage"
+    t.decimal "timeshare_property_value"
+    t.decimal "land_value"
+    t.decimal "valuable_items_value"
+    t.decimal "money_assets_value"
+    t.decimal "money_owed_value"
+    t.decimal "trust_value"
```
Comment (Kalagan) [inline db/schema.rb]: This file contains lots of change irrelevant to this PR.
Doing this should fix it:
1. revert the changes by copying the value of this file from master (https://github.com/ministryofjustice/laa-apply-for-legal-aid/blob/master/db/schema.rb)
2. `bin/rails db:schema:load db:migrate db:seed`

File: features/citizens/citizens.feature [hunk 3]
```diff
@@ -298,4 +298,24 @@ Feature: Citizen journey
     Then I should be on a page showing "What is your relationship to"
     Then I choose "They're a child relative"
     Then I click "Save and continue"
-    Then I should be on a page showing "dependants income"
+    Then I should be on a page showing "receive any income?"
+
+  @javascript
+  Scenario: I have an adult dependant more than 18 years old
```
Comment (Kalagan) [inline features/citizens/citizens.feature]: Rather than creating a new scenario, I suggest carrying on with the previous one.

File: app/forms/dependant_form/monthly_income_form.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
+module DependantForm
+  class MonthlyIncomeForm
+    include BaseForm
+    form_for Dependant
+
+    before_validation :clear_monthly_income
+
+    attr_accessor :has_income, :monthly_income
+
+    validate :income_presence
```
Comment (reggieb) [inline app/forms/dependant_form/monthly_income_form.rb]: Why doesn't this work:
```ruby
validates :has_income, presence: true
```
That would avoid using a custom method.

File: app/services/flow/flows/citizen_dependants.rb (hunk 1 above)
Comment (reggieb) [inline app/services/flow/flows/citizen_dependants.rb]: I think this will work:
```ruby
forward: :depandants_assets
```

File: app/views/shared/forms/dependants/_monthly_incomes.html.erb [hunk 5]
```diff
@@ -0,0 +1,45 @@
+<%= page_template(
+      page_title: t('.page_title', name: @form.model.name),
+      back_link: { text: t('generic.back') },
```
Comment (reggieb) [inline app/views/shared/forms/dependants/_monthly_incomes.html.erb]:
```ruby
back_link: { text: t('generic.back') },
```
You don't need that line. `page_template` defaults to using `t('generic.back')`

File: db/schema.rb [hunk 6]
```diff
@@ -264,23 +266,23 @@
     t.boolean "open_banking_consent"
     t.datetime "open_banking_consent_choice_at"
     t.string "own_home"
-    t.decimal "percentage_home"
     t.decimal "property_value", precision: 10, scale: 2
     t.string "shared_ownership"
     t.decimal "outstanding_mortgage_amount"
+    t.decimal "percentage_home"
```
Comment (reggieb) [inline db/schema.rb]: These fields moving suggest you didn't `rake db:schema:load` on starting this branch. The schema needs to be fixed, so that only the entries made in the migration modify the schema

File: app/forms/dependant_form/monthly_income_form.rb (hunk 4 above)
Comment (AbdirizakObsiye2018) [reply app/forms/dependant_form/monthly_income_form.rb]: I used the custom method because the error message required the dependants name within it e.g. Select yes if John receives any income

File: features/citizens/citizens.feature (hunk 3 above)
Comment (AbdirizakObsiye2018) [reply features/citizens/citizens.feature]: Carrying onto creating a new dependant that's more than 18 years old, will require the add assets page / feature which hasn't been implemented yet ( add income page - > add assets page - > any other dependants page )
